Fortnox.NET.Tests/Fortnox.NET.Tests/AssetsTest.cs

```diff
@@ -72,5 +74,125 @@ public async Task ItCanUpdateAnAsset()
             
             Assert.AreEqual(updatedDescription, result.Description);
         }
+
+        [TestMethod]
+        public async Task ItCanWriteUpAndDownAnAsset()
+        {
+            var currentDate = new DateTime(DateTime.UtcNow.Year, DateTime.UtcNow.Month, 1);
+            var request = new FortnoxApiRequest(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret);
+            var result = await AssetsService.GetAssetAsync(request, "4");
+
+            Assert.AreEqual("2", result.Number);
+
+            var writeUp = new WriteUpOrDownAsset { Amount = 300, Comment = "WriteUp, Possible comment", Date = currentDate.AddMonths(1).ToString("yyyy-MM-dd") };
+            var writeUpAsset = await AssetsService.WriteUpAssetAsync(request, $"{result.Id}", writeUp);
+
+            var latestWriteUp = writeUpAsset.History.Last();
+            Assert.IsTrue(latestWriteUp.Notes.Contains(writeUp.Comment));
+
+            var writeDown = new WriteUpOrDownAsset { Amount = 200, Comment = "WriteDown, Possible comment", Date = currentDate.AddMonths(1).ToString("yyyy-MM-dd") };
+            var writeDownAsset = await AssetsService.WriteDownAssetAsync(request, $"{result.Id}", writeDown);
+
+            var latestWriteDown = writeDownAsset.History.Last();
+            Assert.IsTrue(latestWriteDown.Notes.Contains(writeDown.Comment));
+        }
+
+        [TestMethod]
+        public async Task ItCanScrapAnAsset()
+        {
+
+            var currentDate = new DateTime(DateTime.UtcNow.Year, DateTime.UtcNow.Month, 1);
+            var rand = new Random();
+            var asset = new Asset()
+            {
+                Number = "TestProd" + rand.Next(9999999),
+                Description = "TEST PRODUCT",
+                TypeId = "8",
+                DepreciationMethod = "0",
+                AcquisitionValue = 13000,
+                DepreciateToResidualValue = 1300,
+                AcquisitionDate = currentDate.ToString("yyyy-MM-dd"),
+                AcquisitionStart = currentDate.AddMonths(1).ToString("yyyy-MM-dd"),
+                DepreciationFinal = currentDate.AddYears(1).ToString("yyyy-MM-dd"),
+            };
+
+            var request = new FortnoxApiRequest(this.connectionSettings.AccessToken, this.connectionSettings.ClientSecret);
+            var result = await AssetsService.CreateAssetAsync(request, asset);
+
+            Assert.AreEqual(asset.Number, result.Number);
+
+            var scrapAsset = new AssetScrap { Percentage = 5, Comment = "Scrap comment", Date = currentDate.AddMonths(1).ToString("yyyy-MM-dd") };
+            var scrappedAsset = await AssetsService.ScrapAssetAsync(request, $"{result.Id}", scrapAsset);
+
+            var latestHistoryEntry = scrappedAsset.History.Last();
+            Assert.IsTrue(latestHistoryEntry.Notes.Contains(scrapAsset.Comment));
+
+            await AssetsService.DeleteAssetAsync(request, result.Id.ToString());
```

Comment:
Maybe want to move the delete to before the assert so it gets deleted even if the assert fails.

Reply:
@brokenprogrammer What happens if this isn't executed? Is the underlying dataset affected in such a way that future tests won't pass? If so I agree with @BaconFinger

Reply:
Its based on Dylans Implementation where a random Asset is generated with  `Number = "TestProd" + rand.Next(9999999),` so yes and no to your question @GGAlanSmithee eventually in the future there would be too many generated Assets in the dataset.

Reply:
OK, I see, as a general rule I think it'd be good to keep the tests as predictable as possible, that is "leave it as you find it".

Reply:
Is there anything actionable to do here in regards to the above?

Reply:
Few options:

- Move the delete to before the assert.
- Move the assert (and anything else that may throw) into a try-finally
- Check if any entity exists before creating a new one.

I don't think the last one is reasonable here since 'scrap' sounds close enough to 'delete'. Making slight, inconsequential modifications to existing data is fine IMO, but you don't want to be doing anything too crazy on that data.

Reply:
@BaconFinger @GGAlanSmithee I moved the delete call before the assert on the result from scrapping the Asset.

Reply:
Looks good to me. Dunno how to merge it though. Looks like my only option is to close it?